src/Stratis.SmartContracts.IntegrationTests/ContractExecutionFailureTests.cs

```diff
@@ -738,49 +739,55 @@ public void ContractTransaction_Call_Method_Consumes_All_Gas_Throws_OutOfGasExce
         {
-            const ulong gasPrice = SmartContractMempoolValidator.MinGasPrice;
-            var gasLimit = (Gas)(SmartContractFormatLogic.GasLimitMaximum / 2);
-            const ulong txGasPerBlockLimit = SmartContractFormatLogic.GasLimitMaximum * 10;
-            const int txCount = 25;
-            decimal amount = 0;
+            var gasPrice_100 = SmartContractMempoolValidator.MinGasPrice;
+            var gasLimit_50_000 = (Gas)(SmartContractFormatLogic.GasLimitMaximum / 2);
+            var txGasPerBlockLimit_1_000_000 = SmartContractFormatLogic.GasLimitMaximum * 10;
+            var txCount_25 = 25;
 
             ContractCompilationResult compilationResult = ContractCompiler.CompileFile("SmartContracts/InfiniteLoop.cs");
             Assert.True(compilationResult.Success);
 
-            BuildCreateContractTransactionResponse preResponse = this.node1.SendCreateContractTransaction(compilationResult.Compilation, 0, gasPrice: gasPrice);
+            // Create the contract and mine it.
+            BuildCreateContractTransactionResponse preResponse = this.node1.SendCreateContractTransaction(compilationResult.Compilation, 0, gasPrice: gasPrice_100);
             this.mockChain.WaitAllMempoolCount(1);
             this.mockChain.MineBlocks(1);
+
+            // Ensure that the contract is created.
             Assert.NotNull(this.node1.GetCode(preResponse.NewContractAddress));
 
             var contractTransactionIds = new List<uint256>();
 
-            for (int i = 0; i < txCount; i++)
+            // Create 25 call contract transactions.
+            for (int i = 0; i < txCount_25; i++)
             {
-                BuildCallContractTransactionResponse response =
-                    this.node1.SendCallContractTransaction("Loop", preResponse.NewContractAddress, amount);
+                BuildCallContractTransactionResponse response = this.node1.SendCallContractTransaction("Loop", preResponse.NewContractAddress, 0);
                 if (!response.Success)
                     Assert.True(response.Success);
+
                 contractTransactionIds.Add(response.TransactionId);
             }
 
-            this.mockChain.WaitAllMempoolCount(txCount);
+            // Try and include the 25 transactions in the block (only 20 will be added).
+            this.mockChain.WaitAllMempoolCount(txCount_25);
             this.mockChain.MineBlocks(1);
 
             NBitcoin.Block lastBlock = this.node1.GetLastBlock();
 
-            int expectedTxQty = Convert.ToInt32(txGasPerBlockLimit / gasLimit) + 1; // +1 is Coinbase Tx.
-            Assert.Equal(expectedTxQty, lastBlock.Transactions.Count);
+            // There should be 20 transactions in the last block as that is when the block gas expenditure limit was reached.
+            // 19 included transactions plus 1 coinbase.
+            int expectedTxCount_20 = Convert.ToInt32(txGasPerBlockLimit_1_000_000 / gasLimit_50_000);
```

Comment:
@codingupastorm can you give some special attention here to ensure that the test values are asserted correctly?

Reply:
Because they're all 50000 limit, we should let all of them in, so the total in the block should actually be 21, for all 20 and the coinbase. 

I think this will be fixed with the suggestion I made in the coinview rule.